File: python/leetcode187.py
class Solution(object):
    def findRepeatedDnaSequences(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        d = {}
        for i in range(len(s)-9):
            d[s[i:i+10]] = d.get(s[i:i+10],0)+1
        return [key for key in d if d[key] >1]

File: leetcodec++/leetcode187.cpp
//
// Created by 黄敏辉 on 2019-08-26.
//

class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        vector<string> res;
        int l = s.size();
        unordered_map<string, int> m;
        for (int i = 0; i < l;i++){
            string sub = s.substr(i, 10);
            if(m[sub] == 1)
                res.push_back(sub);
            m[sub] ++;
        }

        return res;
    }
};